### PR TITLE
Add example and documentation on multicategorical

### PR DESCRIPTION
**Pull request by yiweny**

### CODE REVIEW

**Review by weihua916 — APPROVED** [review]
Looks great. Curious about the model performance.

File: examples/mercari.py [hunk 1]
```diff
@@ -1,9 +1,168 @@
...
+
+torch.manual_seed(args.seed)
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+# Prepare datasets
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'mercari')
+text_encoder = PretrainedTextEncoder(device=device)
+dataset = Mercari(
+    root=path,
+    num_rows=5000,  # delete the argument to use the full dataset
```
Comment (weihua916) [inline examples/mercari.py:54]: Can we remove this and report performance against public leaderboard before merging?

File: examples/mercari.py [hunk 2]
```diff
@@ -1,9 +1,174 @@
...
+    FTTransformer,
+    LinearEmbeddingEncoder,
+    LinearEncoder,
+)
+from torch_frame.nn.encoder.stype_encoder import \
+    MultiCategoricalEmbeddingEncoder  # noqa
+
+
+class PretrainedTextEncoder:
+    def __init__(self, device: torch.device):
```
Comment (akihironitta) [inline examples/mercari.py:26]: nit: https://peps.python.org/pep-0484/#the-meaning-of-annotations
```suggestion
    def __init__(self, device: torch.device) -> None:
```

File: examples/mercari.py [hunk 3]
```diff
@@ -1,9 +1,174 @@
...
+
+torch.manual_seed(args.seed)
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+# Prepare datasets
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'mercari')
+text_encoder = PretrainedTextEncoder(device=device)
+dataset = Mercari(
+    root=path,
+    # num_rows=5000,  # delete the argument to use the full dataset
```
Comment (akihironitta) [inline examples/mercari.py:54]:
```suggestion
    # num_rows=5000,  # set num_rows to use a subset of the dataset
```

**Review by akihironitta — APPROVED** [review]
🚀

File: submission.csv [hunk 4]
```diff
@@ -0,0 +1,1001 @@
+,test_id,price
```
[the remark this answers is not in the document]
Comment (yiweny) [reply submission.csv:1]: Ops. Let me fix this. I have no plan to merge it now because it ooms anyways.

**Pull request merged**